### PR TITLE
Document FSx

### PR DESCRIPTION
**Pull request by sean-smith**
### Overview

* Updates configuration.rst with new parameters
* Updates EBS KMS Key Tutorial to make it a more general KMS Tutorial.

### Resources
1. [CreateFileSystem Docs](https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystem.html)
2. [Lustre Configuration Docs](https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystemLustreConfiguration.html)

Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
...
+Create a new file system, by omitting ``fsx_fs_id``, or use an existing one by specifying ``fsx_fs_id``. ::
+
+    [fsx fs]
+    shared_dir = /fsx
+    fsx_fs_id = fs-073c3803dca3e28a6
+    storage_capacity = 3600
+    fsx_kms_key_id = 9e8a129c-0e85-459-865b-3a5be974a22b
+    imported_file_chunk_size = 1024
+    export_path = s3://bucket/folder
+    import_path =  s3://bucket
```
Comment (enrico-usai) [inline docs/configuration.rst]: minor: extra space

File: docs/configuration.rst [hunk 2]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
...
+"""""""""
+**Optional** Attach an existing FSx File System.
+
+If this option is specified, all following FSx parameters, such as ``storage_capacity`` are ignored. ::
+
+    fsx_fs_id = fs-073c3803dca3e28a6
+
+storage_capacity
+""""""""""""""""
+**Optional** The storage capacity of the file system.
```
Comment (enrico-usai) [inline docs/configuration.rst]: (in GiB)

File: docs/configuration.rst (hunk 1 above)
Comment (sean-smith) [reply docs/configuration.rst]: removed

File: docs/configuration.rst (hunk 2 above)
Comment (sean-smith) [reply docs/configuration.rst]: Added

File: docs/tutorials/04_encrypted_ebs.rst [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
-###################################
-Encrypted EBS with a Custom KMS Key
-###################################
+#####################################
+Disk Encryption with a Custom KMS Key
+#####################################
 
-AWS ParallelCluster supports the configuration option ``ebs_kms_key_id``, which allows you to provide a custom KMS key
-for EBS Disk encryption, to use it you'll need to specify a ``ec2_iam_role``.
+AWS ParallelCluster supports the configuration option ``ebs_kms_key_id`` and ``fsx_kms_key_id``, which allows you to
```
Comment (demartinofra) [inline docs/tutorials/04_encrypted_ebs.rst]: configuration options...which allow...custom KMS key for EBS Disk (FSx is missing)

File: docs/configuration.rst [hunk 4]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
...
+
+.. _fsx_section:
+
+FSx
+^^^
+Configuration for an attached FSx Lustre file system.
+
+Create a new file system, by omitting ``fsx_fs_id``, or use an existing one by specifying ``fsx_fs_id``. ::
+
+    [fsx fs]
```
Comment (demartinofra) [inline docs/configuration.rst]: Shall we have 2 separate examples: one with fs_id and one without to better highlight what are the parameters that are actually used in each case? Cause I expect users to simply copy paste this section and ignoring the rest :)

File: docs/configuration.rst [hunk 5]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
...
+    import_path = s3://bucket
+    weekly_maintenance_start_time = 1:00:00
+
+shared_dir
+""""""""""
+**Required** Defines the mount point for the Lustre File system on the master and compute nodes.
+
+The example below will mount the filesystem at /fsx.
+
+Do not use NONE or /NONE as the shared directory.::
```
Comment (demartinofra) [inline docs/configuration.rst]: Side question: do we validate this?

File: docs/tutorials/04_encrypted_ebs.rst [hunk 6]
```diff
@@ -3,12 +3,12 @@
 .. toctree::
    :maxdepth: 2
 
-###################################
-Encrypted EBS with a Custom KMS Key
-###################################
+#####################################
+Disk Encryption with a Custom KMS Key
```
Comment (demartinofra) [inline docs/tutorials/04_encrypted_ebs.rst]: Shall we modify the filename as well?

File: docs/configuration.rst [hunk 7]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
...
+
+    storage_capacity = 3600
+
+fsx_kms_key_id
+""""""""""""""
+**Optional** The ID of your AWS Key Management Service (AWS KMS) key.
+
+This ID is used to encrypt the data in your file system at rest.
+
+This must be used with a custom ``ec2_iam_role``. See `Encrypted EBS with a Custom KMS Key <_encrypted_ebs>`. ::
```
Comment (demartinofra) [inline docs/configuration.rst]: This linked section has now a different name

File: docs/configuration.rst [hunk 8]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
...
+the ``import_path`` parameter.
+
+Defaults to ``s3://import-bucket/FSxLustre[creation-timestamp]`` where ``import-bucket`` is the bucket provided in
+``import_path`` parameter. ::
+
+    export_path = s3://bucket/folder
+
+import_path
+"""""""""""
+***Optional** S3 Bucket to load data from into the file system. Also serves as the export bucket. See ``export_path``.
```
Comment (demartinofra) [inline docs/configuration.rst]: Might be worth specifying that import happens only at filesystem, hence cluster, creation time

File: docs/configuration.rst [hunk 9]
```diff
@@ -971,3 +971,100 @@ This must be used in conjunction with ``encrypted = true`` and must have a custo
 See `Encrypted EBS with a Custom KMS Key <_encrypted_ebs>`. ::
 
     ebs_kms_key_id = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+
+
+.. _fsx_section:
+
+FSx
```
Comment (demartinofra) [inline docs/configuration.rst]: Shall we add a link to the official FSx docs at the end of this section?

Comment (demartinofra) [reply docs/configuration.rst]: `fsx_settings` option from the cluster section doesn't seem to be documented

File: docs/configuration.rst (hunk 4 above)
Comment (sean-smith) [reply docs/configuration.rst]: good idea

File: docs/configuration.rst (hunk 5 above)
Comment (sean-smith) [reply docs/configuration.rst]: That it's not `NONE` or `/NONE`? We don't.

File: docs/tutorials/04_encrypted_ebs.rst (hunk 6 above)
Comment (sean-smith) [reply docs/tutorials/04_encrypted_ebs.rst]: Modified `04_encrypted_ebs.rst -> 04_encrypted_kms_fs.rst`

File: docs/tutorials/04_encrypted_ebs.rst (hunk 3 above)
Comment (sean-smith) [reply docs/tutorials/04_encrypted_ebs.rst]: Added

**Pull request merged**